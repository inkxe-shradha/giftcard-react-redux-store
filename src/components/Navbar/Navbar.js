import { useTheme } from "@emotion/react";
import {
  Divider,
  Drawer,
  IconButton,
  List,
  ListItemText,
  ListItemButton,
  ListItemIcon,
  Paper,
} from "@mui/material";

import React from "react";
import ChevronRightIcon from "@mui/icons-material/ChevronRight";
import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
import menuItems from "../../config/menuList";
import { useLocation, useNavigate } from "react-router-dom";
const Navbar = ({
  drawer,
  open,
  handleDrawerClose,
  toolbar,
  drawerPaper,
  navDrawer,
}) => {
  const theme = useTheme();
  const navigate = useNavigate();
  const location = useLocation();
  return (
    <nav className={navDrawer}>
      <Paper
        sx={{ display: { xs: "none", sm: "block", md: "block" } }}
        className={drawerPaper}
      >
        <Drawer
          variant="temporary"
          elevation={0}
          open={open}
          className={drawer}
          onClose={handleDrawerClose}
          classes={{
            paper: drawerPaper,
          }}
          anchor={theme.direction === "rtl" ? "right" : "left"}
          ModalProps={{
            keepMounted: true, // Better open performance on mobile.
          }}
        >
          <div className={toolbar}>
            <IconButton onClick={handleDrawerClose}>
              {theme.direction === "rtl" ? (
                <ChevronRightIcon />
              ) : (
                <ChevronLeftIcon />
              )}
            </IconButton>
          </div>
          <Divider />
          <List>
            {menuItems.map((item) => (
              <ListItemButton
                key={item.id}
                selected={location.pathname === item.path}
                onClick={() => navigate(item.path)}
              >
                <ListItemIcon>{item.icon}</ListItemIcon>
                <ListItemText primary={item.name} />
              </ListItemButton>
            ))}
          </List>
        </Drawer>
      </Paper>
    </nav>
  );
};

export default Navbar;
