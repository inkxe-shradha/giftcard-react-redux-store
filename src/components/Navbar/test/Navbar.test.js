import { render } from "@testing-library/react";
import Navbar from "../Navbar";
import { BrowserRouter } from "react-router-dom";

describe("Navbar Test Suite Testing", () => {
  it("Should able to capture the snapshot of the Navbar", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navbar open={true} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
