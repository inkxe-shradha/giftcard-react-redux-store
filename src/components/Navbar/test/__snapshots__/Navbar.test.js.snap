// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Test Suite Testing Should able to capture the snapshot of the Navbar 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-joixkg-MuiPaper-root"
    />
  </nav>
</DocumentFragment>
`;
